nnvm/src/top/nn/nn.cc

```diff
@@ -381,7 +384,7 @@ NNVM_REGISTER_OP(softmax)
     NodeEntry sub2 = MakeNode("broadcast_mul", n->attrs.name + "_grad_sub2", {sub1, output});
     NodeEntry sub3 = MakeNode("negative", n->attrs.name + "_grad_sub3", {sub2});
```

Comment:
what happens if you remove this and replace the following node with `elemwise_sub`? Although op fusion is possible during the backward pass, this isn't yet implemented. Might as well prevent one temporary :)